R/create_epa.R

```diff
@@ -155,199 +183,131 @@ create_epa <- function(clean_pbp_dat,
       pred_df[kickoff_ind,"ep_before"] = sum(ep_kickoffs * weights)
     }
   }
-
+  
+  # **Due to ESPN data quality issues, some drives end on 3rd down that are listed as turnovers
   # For turnover and punt plays make sure the ep_after is negative
-  # because of poor ESPN data quality,
-  # some drives end on 3rd down and we have those listed as turnovers
-  turnover_plays = which(pred_df$turnover == 1 & !kickoff_ind & pred_df$play_type %in% turnover_play_type)
-  pred_df[turnover_plays, "ep_after"] = -1 * pred_df[turnover_plays, "ep_after"]
-
+  # 
+  turnover_plays = which(pred_df$turnover == 1 & !kickoff_ind & (pred_df$play_type %in% turnover_play_type))
+  pred_df[turnover_plays, "ep_after"] = -1*pred_df[turnover_plays, "ep_after"]
+  
   # game end EP is 0
   pred_df[pred_df$end_half_game_end == 1, "ep_after"] = 0
 
   ## scoring plays from here on out
   pred_df[(pred_df$play_type %in% off_TD), "ep_after"] = 7
   pred_df[(pred_df$play_type %in% def_TD), "ep_after"] = -7
+  pred_df[pred_df$play_type == "Defensive 2pt Conversion", "ep_before"] = 1
```

Comment:
Why are you setting the ep_before here as 1?

Reply:
These are specific play_types with defensive 2pt conversion as the play_type, meaning they're separate from the touchdown play. I'm basically equating the expectation of a point after as 1 (could be like .96 or whatever). I am planning on adjusting for PATs shortly to give specific ep_afters based on whether the PAT was successful a couple lines up. I'm planning on getting to that this week

Reply:
Oh wait, should I be setting it to zero and just not worry about that? I think that'd kinda fix it other than giving extra credit to the offense on those 6 prior-play touchdowns.

Reply:
Yeah. I might lean towards the 0 solution. Because the pbp by doesn't separate out kickoffs. (This is something we can look to tackle more long term say in the off-season).